crates/polars-core/src/series/arithmetic/list_borrowed.rs

```diff
@@ -53,23 +53,112 @@ fn lists_same_shapes(left: &ArrayRef, right: &ArrayRef) -> bool {
     }
 }
 
+/// Arithmetic operations that can be applied to a Series
+#[derive(Clone, Copy)]
+enum Op {
+    Add,
+    Subtract,
+    Multiply,
+    Divide,
+    Remainder,
+}
+
+impl Op {
+    /// Apply the operation to a pair of Series.
+    fn apply_with_series(&self, lhs: &Series, rhs: &Series) -> PolarsResult<Series> {
```

Comment:
I tried to do these two functions as a single generic function and failed...

Reply:
Why exactly? Seems this is possible with generics.

Reply:
I couldn't figure out the constraints that work. I guess I can try another round and report back when I get stuck.

Reply:
So the issue is that when you add two `&Series`, you get a `PolarsResult<Series>`, and when you add a `&Series` and `u8` you get `Series`. Since the return types don't match, the constraints conflict (the former wants `T: std:ops::Add<Output=PolarsResult<Series>>`, the latter wants `T: std:ops::Add<Output=Series>`).